v21.2/deploy-lambda-function.md

```diff
@@ -0,0 +1,164 @@
+
+Open a terminal window and copy the sample code's GitHub repo:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/examples-aws-lambda/
+~~~
+
+The function's code is available under the `examples-aws-lambda/python` directory:
+
+~~~ shell
+.
+├── README.md
+├── deployment-package.zip  ## Lambda deployment package
+├── init_db.py              ## Lambda function source code
+├── package                 ## Psycopg dependencies
+├── requirements.txt        ## List of Python requirements
+└── root.crt                ## CA cert
+~~~
+
+## Step 3. (Optional) Create the deployment package
+
+{{site.data.alerts.callout_info}}    
+This step is optional, as you do not need to create a new deployment package to deploy the sample function. The `examples-aws-lambda` repo includes a deployment package that is ready to deploy.
+{{site.data.alerts.end}}
+
+1. Download and install the `psycopg2-binary` Python library to a new directory:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ python3 -m pip install --only-binary :all: --platform manylinux1_x86_64  --target ./my-package -r requirements.txt
+    ~~~
+
+    {{site.data.alerts.callout_info}}    
+    To run on Amazon Linux distributions, `pscyopg2` dependencies must be compiled for Linux.
+    {{site.data.alerts.end}}
+
+1. Compress the project files to a ZIP file for deployment:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ cd my-package
+    ~~~
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ zip -r ../my-deployment-package.zip .
+    ~~~
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    cd ..
+    ~~~
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    zip -g my-deployment-package.zip init_db.py root.crt
+    ~~~
+
+## Step 4. Configure AWS
+
+1. Configure the AWS CLI to authenticate with your AWS account:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ aws configure
+    ~~~
+
+    Follow the prompts to authenticate.
+
+1. Create an execution role for the Lambda function and attach the `AWSLambdaBasicExecutionRole` policy to the role:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ aws iam create-role --role-name lambda-ex --assume-role-policy-document '{"Version": "2012-10-17","Statement": [{ "Effect": "Allow", "Principal": {"Service": "lambda.amazonaws.com"}, "Action": "sts:AssumeRole"}]}'
+    ~~~
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ aws iam attach-role-policy --role-name lambda-ex --policy-arn arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+    ~~~
+
+    The Lambda function needs this role to run.
+
+## Step 5. Deploy the function to AWS Lambda
+
+1. In the deployment package directory, use the AWS CLI to create a Lambda function:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ aws lambda create-function \
```

Comment:
Is there a sane way to make this happen without leaking the DB URL into the shell history? I can't really think of one.